spec/rubocop/config_spec.rb

```diff
@@ -655,9 +666,38 @@ def cop_enabled(cop_class)
         }
       end
 
-      it 'uses the default rails version' do
-        default_version = RuboCop::Config::DEFAULT_RAILS_VERSION
-        expect(configuration.target_rails_version).to eq default_version
+      context 'and lock files are missing' do
+        it 'uses the default rails version' do
+          default = RuboCop::Config::DEFAULT_RAILS_VERSION
+          expect(configuration.target_rails_version).to eq default
+        end
+      end
+
+      ['Gemfile.lock', 'gems.locked'].each do |file_name|
+        context "and #{file_name} exists" do
+          let(:base_path) { configuration.base_dir_for_path_parameters }
+          let(:lock_file_path) { File.join(base_path, file_name) }
+
+          before do
+            allow(configuration).to(
+              receive(:bundler_lock_file_name).and_return(lock_file_path)
+            )
```

Comment:
Probably you can use `isolated_environment` instead of the mock.
https://github.com/bbatsov/rubocop/blob/698bd2ee06aa365d85850a2ff23f71c46e0fab06/lib/rubocop/rspec/shared_contexts.rb#L6
it creates a temporary directory for the test.
I think that we should avoid mock as much as possible.